Sources/HTMLEntities/String+HTMLEntities.swift

```diff
@@ -0,0 +1,338 @@
+    "&harr;":"\u{2194}","&crarr;":"\u{21B5}","&lArr;":"\u{21D0}","&uArr;":"\u{21D1}",
+    "&rArr;":"\u{21D2}","&dArr;":"\u{21D3}","&hArr;":"\u{21D4}","&forall;":"\u{2200}",
+    "&part;":"\u{2202}","&exist;":"\u{2203}","&empty;":"\u{2205}","&nabla;":"\u{2207}",
+    "&isin;":"\u{2208}","&notin;":"\u{2209}","&ni;":"\u{220B}","&prod;":"\u{220F}",
+    "&sum;":"\u{2211}","&minus;":"\u{2212}","&lowast;":"\u{2217}","&radic;":"\u{221A}",
+    "&prop;":"\u{221D}","&infin;":"\u{221E}","&ang;":"\u{2220}","&and;":"\u{2227}",
+    "&or;":"\u{2228}","&cap;":"\u{2229}","&cup;":"\u{222A}","&int;":"\u{222B}",
+    "&there4;":"\u{2234}","&sim;":"\u{223C}","&cong;":"\u{2245}","&asymp;":"\u{2248}",
+    "&ne;":"\u{2260}","&equiv;":"\u{2261}","&le;":"\u{2264}","&ge;":"\u{2265}",
+    "&sub;":"\u{2282}","&sup;":"\u{2283}","&nsub;":"\u{2284}","&sube;":"\u{2286}",
+    "&supe;":"\u{2287}","&oplus;":"\u{2295}","&otimes;":"\u{2297}","&perp;":"\u{22A5}",
+    "&sdot;":"\u{22C5}","&lceil;":"\u{2308}","&rceil;":"\u{2309}","&lfloor;":"\u{230A}",
+    "&rfloor;":"\u{230B}","&lang;":"\u{2329}","&rang;":"\u{232A}","&loz;":"\u{25CA}",
+    "&spades;":"\u{2660}","&clubs;":"\u{2663}","&hearts;":"\u{2665}","&diams;":"\u{2666}",
+
+    // Markup-significant and internationalization characters
+    // https://www.w3.org/TR/html4/sgml/entities.html#h-24.4.1
+    "&quot;":"\u{0022}","&amp;":"\u{0026}","&lt;":"\u{003C}","&gt;":"\u{003E}",
+    "&OElig;":"\u{0152}","&oelig;":"\u{0153}","&Scaron;":"\u{0160}","&scaron;":"\u{0161}",
+    "&Yuml;":"\u{0178}","&circ;":"\u{02C6}","&tilde;":"\u{02DC}","&ensp;":"\u{2002}",
+    "&emsp;":"\u{2003}","&thinsp;":"\u{2009}","&zwnj;":"\u{200C}","&zwj;":"\u{200D}",
+    "&lrm;":"\u{200E}","&rlm;":"\u{200F}","&ndash;":"\u{2013}","&mdash;":"\u{2014}",
+    "&lsquo;":"\u{2018}","&rsquo;":"\u{2019}","&sbquo;":"\u{201A}","&ldquo;":"\u{201C}",
+    "&rdquo;":"\u{201D}","&bdquo;":"\u{201E}","&dagger;":"\u{2020}","&Dagger;":"\u{2021}",
+    "&permil;":"\u{2030}","&lsaquo;":"\u{2039}","&rsaquo;":"\u{203A}","&euro;":"\u{20AC}"
+]
+
+// constants used to avoid character initializations in loop
+private let ampersand: Character = "&"
+private let semicolon: Character = ";"
+
+/// This String extension provides utility functions to convert strings to their
+/// HTML named character entity equivalents and vice versa. Named character
+/// entities are often used to mitigate XSS attacks that, for example, inject
+/// scripts into web pages.
+public extension String {
+    /// Return string as HTML encoded by replacing certain characters with their
+    /// HTML named character entity equivalents.
+    /// For example, this function turns
+    /// `"<script>alert(\"abc\")</script>"`
+    /// into
+    /// `"&lt;script&gt;alert(&quot;abc&quot;)&lt;/script&gt;"`
+    public func asHTMLEntities() -> String {
+        var str = ""
+
+        for char in self.characters {
+            if let entity = html4encode[char] {
+                str.append(entity)
```

Comment:
lazy instantiation of str on the first occurrence of a mapped character would be a lot more efficient, especially in typical use cases where the vast majority of characters do not need escaping. If no characters need mapping we could return the original string. Same thing applies to the reverse function below.